docs/user-guide/source-hydrator.md

```diff
@@ -147,6 +147,75 @@ Argo CD will only push changes to the `hydrateTo` branch, it will not create a P
 changes to the `syncSource` branch. You will need to use your own tooling to move the changes from the `hydrateTo` 
 branch to the `syncSource` branch.
 
+## Commit Tracing
+
+It's common for CI or other tooling to push DRY manifest changes after a code change. It's important for users to be
+able to trace the hydrated commits back to the original code change that caused the hydration.
+
+Source Hydrator makes use of some custom git commit trailers to facilitate this tracing. A CI job that builds an image
+and pushes an image bump to DRY manifests can use the following commit trailers to link the hydrated commit to the
+code commit.
+
+```shell
+git commit -m "Bump image to v1.2.3" \
+  --trailer "Argocd-related-commit-author: Author Name <author-email>" \
+  --trailer "Argocd-related-commit-sha: <code-commit-sha>" \
+  --trailer "Argocd-related-commit-subject: Commit message of the code commit" \
+  --trailer "Argocd-related-commit-repourl: https://git.example.com/owner/repo" \
+  --trailer "Argocd-related-commit-date: 2025-06-09T13:50:18-04:00" # The date must by in ISO 8601 format
+```
+
+So the full CI script might look something like this:
+
+```shell
+# Clone code repo
+git clone https://git.example.com/owner/repo.git
+cd repo
+
+# Build the image and get the new image tag
+# <cusom build logic here>
+
+# Get the commit information
+author=$(git show -s --format='%an <%ae>')
+sha=$(git rev-parse HEAD)
+message=$(git show -s --format='%s')
+repourl=$(git remote get-url origin)
+date=$(git show -s --format='%aI')
+
+# Clone the dry source repo
+git clone https://git.example.com/owner/deployment-repo.git
+cd deployment-repo
+
+# Bump the image in the dry manifests
+# <custom bump logic here, e.g. `kustomize edit`>
+
+# Commit the changes with the commit trailers
+git commit -m "Bump image to v1.2.3" \
+  --trailer "Argocd-related-commit-author: $author" \
+  --trailer "Argocd-related-commit-sha: $sha" \
+  --trailer "Argocd-related-commit-subject: $message" \
+  --trailer "Argocd-related-commit-repourl: $repourl" \
+  --trailer "Argocd-related-commit-date: $date"
+```
+
+The commit metadata will appear in the hydrated commit's root hydrator.metadata file:
+
+```json
+{
+  "relatedCommits": [
+    {
+      "author": "Author Name <author-email>",
+      "sha": "<code-commit-sha>",
+      "message": "Commit message of the code commit",
+      "repoURL": "https://git.example.com/owner/repo",
+      "date": "2025-06-09T13:50:18-04:00"
+    }
+  ]
+}
```

Comment:
By default, support (or not) Argocd-related-commit-type as a header. If it is not specifid, hydrator can decide to always assume git

```suggestion
{
  "references": [
    {
      "type": "commit"
      "author": "Author Name <author-email>",
      "sha": "<code-commit-sha>",
      "message": "Commit message of the code commit",
      "repoURL": "https://git.example.com/owner/repo",
      "date": "2025-06-09T13:50:18-04:00"
    }
  ]
}
```